contentreview-impl/impl/src/java/org/sakaiproject/contentreview/impl/turnitin/TurnitinReviewServiceImpl.java

```diff
@@ -3425,14 +3344,49 @@ private String switchLTIError(int result, String method){
 				return method + ": generic LTI error";
 		}
 	}
-	
-	private void processError(ContentReviewItem item, Long status, String error){//TODO resto de llamadas
-		item.setStatus(status);
-		if(error != null){
+
+	private void processError( ContentReviewItem item, Long status, String error, Integer errorCode )
+	{
+		if( status == null )
+		{
+			IllegalArgumentException ex = new IllegalArgumentException( "Status is null; you must supply a valid status to update when calling processError()" );
+			log.warn( ex );
+			throw ex;
+		}
+		else
+		{
+			item.setStatus( status );
+		}
+		if( error != null )
+		{
 			item.setLastError(error);
 		}
-		dao.update(item);
-		releaseLock(item);
+		if( errorCode != null )
+		{
+			item.setErrorCode( errorCode );
+		}
+
+		dao.update( item );
+		releaseLock( item );
+
+		// Update urlAccessed to true if status is being updated to one of the dead states (5, 6, 8 and 9)
+		if( status.equals( ContentReviewItem.SUBMISSION_ERROR_NO_RETRY_CODE ) || status.equals( ContentReviewItem.REPORT_ERROR_NO_RETRY_CODE ) 
```

Comment:
We really should have `enum`s here and then have the logic about what is a dead state on the enum.

I'd extract this out to a method like `boolean isDead(int status)`